api/routes/units.js

```diff
@@ -77,32 +86,95 @@ router.patch('/:id', async (req, res, next) => {
             response
         })
 
-    } catch {
-
+    } catch (e) {
+        const error = new Error('ID does not match any unit')
+        error.status = 404
+        next(error)
     }
 
 })
 
-// PATCH /api/v1/units/[id]/company
+// [X] PATCH /api/v1/units/[id]/company
 
-// DELETE /api/v1/units/[id]/company - Still Working on
+router.patch('/:id/company', async (req, res, next) => {
+    try {
+        if (req.body.company) {
+            const status = 200
+            const response = await Unit.findOneAndUpdate({
+                _id: req.params.id
+            }, {
+                company: req.body.company
+            }, {
+                new: true
+            })
+            res.json({
+                status,
+                response
+            })
+            // const unit = await Unit.findById(req.params.id)
+            // Object.assign(unit, req.body)
+            // console.log(req.body)
+        }
+    } catch (e) {
+        const error = new Error('ID does not match any unit')
+        error.status = 404
+        next(error)
+    }
+})
+
+// [X] DELETE /api/v1/units/[id]/company - Still Working on
 
 router.delete('/:id/company', async (req, res, next) => {
-    Unit.findOneAndDelete({
-        _id: req.params.id
-    }).then(response => {
-        const status = `Removed compnay with ID ${req.params.id} from the database`
-        res.status(201).json({
+    try {
+        const status = 200
+        const unit = await Unit.findById(req.params.id).select('-__v')
+        console.log(unit)
+        unit.company.remove()
+        console.log(unit)
```

Comment:
You'll want to save the unit (i.e. `unit.save()`) before responding so that the changes take effect.